src/python/bcc/table.py

```diff
@@ -458,6 +460,41 @@ def items_delete_batch(self, keys=None):
             for _ in self.items_lookup_and_delete_batch():
                 return
 
+    def items_update_batch(self, keys, values):
+        """Update all the key-value pairs in the map provided.
+        The lists must be the same length, between 1 and the maximum number of entries.
+        """
+        # two lists are expected
+        if type(keys) != list or type(values) != list:
+            raise TypeError
+
+        batch_size = len(keys)
+
+        # check that batch between limits and coherent with the provided values
+        if (batch_size < 1 and batch_size > self.max_entries) or (batch_size != len(values)):
+            raise KeyError
+        
+        # build the array aka list of key_t that will be deleted
+        keylist = (type(self.Key()) * batch_size)()
+        valuelist = (type(self.Leaf()) * batch_size)()
+        for i, k, v in enumerate(zip(keys, values)):
```

Comment:
I am not sure if this for loop is needed, as for the *items_delete_batch* function.
* Isn't better from the performance point of view that the user already provides the arrays correctly formatted?
* What if the data type instead of being a primitive one (e.g., ct.c_uint32) is a user-defined struct?

Reply:
This is a good point. it is possible that batch_size could be big, so the library here may do double work if user already prepared an array. This is a library function, we can enforce the input parameters to be an array. The same applies to items_delete_batch(). Thanks!

Reply:
No problem :)